contracts/Governance/TokenRedeemer.sol

```diff
@@ -3,63 +3,169 @@ pragma solidity 0.8.13;
 
 import { ReentrancyGuard } from "@openzeppelin/contracts/security/ReentrancyGuard.sol";
 import { Ownable2Step } from "@openzeppelin/contracts/access/Ownable2Step.sol";
-import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
-import { SafeERC20 } from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
 import { ensureNonzeroAddress } from "@venusprotocol/solidity-utilities/contracts/validators.sol";
 
-import { IVBep20 } from "../InterfacesV8.sol";
+import { IVToken, IVBep20, IVBNB } from "../InterfacesV8.sol";
+import { Currency, CurrencyLibrary } from "../lib/Currency.sol";
 
 contract TokenRedeemer is ReentrancyGuard, Ownable2Step {
-    using SafeERC20 for IERC20;
+    using CurrencyLibrary for Currency;
 
+    struct AccountBorrows {
+        address borrower;
+        uint256 amount;
+    }
+
+    struct Borrows {
+        uint256 totalBorrows;
+        AccountBorrows[] accountBorrows;
+    }
+
+    IVBNB public immutable VBNB;
+
+    error AccrueInterestFailed(uint256 errCode);
     error RedeemFailed(uint256 errCode);
     error RepaymentFailed(uint256 errCode);
+    error NativeTokenTransferFailed();
 
-    constructor(address owner_) {
+    constructor(address owner_, IVBNB vBNB) {
         ensureNonzeroAddress(owner_);
+        VBNB = vBNB;
         _transferOwnership(owner_);
     }
 
-    function redeemAndTransfer(IVBep20 vToken, address destination) external nonReentrant onlyOwner {
-        IERC20 underlying = IERC20(vToken.underlying());
+    receive() external payable {}
```

Comment:
I'll add the require, thanks

Reply:
🤔 🤔 🤔

Now I think that I'd rather keep it open. The reason behind this is that we may want to repay someone's borrows with plain BNBs rather than vBNBs, similar to how this contract supports repaying with plain tokens rather than vTokens.

e.g. imagine a situation where we have 1 BNB in vBNB + 5 BNB in BNB. We would like to repay someone's borrow for 3 BNB. If I add the require, we'll need to supply BNB to the vBNB market first, and then redeem them in the same transaction to repay the debt.

Since we have `sweepTokens()`, I think allowing BNB transfers should be ok.